lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -106,20 +107,51 @@ def clean(self):
         cleaned_data['user'] = self._clean_requested_user(cleaned_data, usage_key.course_key)
         return cleaned_data
 
+    def clean_username(self):
+        if 'username' in self.data:
+            return self.cleaned_data['username']
```

Comment:
@xitij2000, a request to http://localhost:18000/api/courses/v2/blocks/?course_id=course-v1%3AedX%2BDemoX%2BDemo_Course&depth=all&username= as a non-staff user in the current `master` branch returns a `"This field is required unless all_blocks is requested."` error. However with this change, the API returns `"'audit' does not have permission to access view for ''"`

Reply:
This is by design. This is the new behaviour. It is not backwards-compatible to this extent. While I could add a special case and have it say <anonymous user> instead of a blank string, I think the current behaviour is also OK, so there is no need for additional complexity. 

While earlier a "username=" was considered an invalid/blank input, it is no longer considered that, so the old error is no longer valid.

Reply:
@xitij2000, I don't have the permission to mark this as resolved. Can you do that?